lib/ex_doc/doc_ast.ex

```diff
@@ -65,6 +65,68 @@ defmodule ExDoc.DocAST do
     Enum.map(attrs, fn {key, val} -> " #{key}=\"#{ExDoc.Utils.h(val)}\"" end)
   end
 
+  @doc """
+  Transform AST into markdown string.
+  """
+  def to_markdown(ast)
+
+  def to_markdown(binary) when is_binary(binary) do
+    ExDoc.Utils.h(binary)
+  end
+
+  def to_markdown(list) when is_list(list) do
+    Enum.map_join(list, "", &to_markdown/1)
+  end
+
+  def to_markdown({:comment, _attrs, inner, _meta}) do
+    "<!--#{inner}-->"
+  end
+
+  def to_markdown({:code, attrs, inner, _meta}) do
+    lang = attrs[:class] || ""
+
+    """
+    ```#{lang}
+    #{inner}
+    ```
+    """
+  end
+
+  def to_markdown({:a, attrs, inner, _meta}) do
+    "[#{to_markdown(inner)}](#{attrs[:href]})"
+  end
+
+  def to_markdown({:hr, _attrs, _inner, _meta}) do
+    "\n\n---\n\n"
+  end
+
+  def to_markdown({:p, _attrs, inner, _meta}) do
+    to_markdown(inner) <> "\n\n"
+  end
+
+  def to_markdown({:br, _attrs, _inner, _meta}) do
+    "\n\n"
+  end
+
+  def to_markdown({:img, attrs, _inner, _meta}) do
+    alt = attrs[:alt] || ""
+    title = attrs[:title] || ""
+    "![#{alt}](#{attrs[:src]} \"#{title}\")"
+  end
+
+  # Ignoring these: area base col command embed input keygen link meta param source track wbr
```

Comment:
Self explanatory code.

```suggestion
```

Alternatively

```suggestion
  # Ignore void elements.
```